BookingSystem/BookingSystem.Queries/Queries/HotelQueries/Queries/HotelDetailsQuery.cs

```diff
@@ -0,0 +1,149 @@
+                            join country in _dataContext.Countries
+                                on hotel.CountryId equals country.CountryId
+                            where hotel.HotelId == query.HotelId
+                            join service in _dataContext.ExtraServices
+                                on hotel.HotelId equals service.HotelId into services
+                            from service in services.DefaultIfEmpty()
+                            group service by new
+                            {
+                                hotel.HotelId,
+                                hotel.Name,
+                                hotel.Address,
+                                CountryName = country.Name,
+                                CityName = city.Name,
+                                hotel.IsActive,
+                            }
+                            into gr
+                            select new HotelDetailsView
+                            {
+                                HotelId = gr.Key.HotelId,
+                                Name = gr.Key.Name,
+                                Address = gr.Key.Address,
+                                CountryName = gr.Key.CountryName,
+                                CityName = gr.Key.CityName,
+                                IsActive = gr.Key.IsActive,
+                                ExtraServices = gr.Select(s => new ExtraServiceView()
+                                {
+                                    ExtraServiceId = s.ExtraServiceId,
+                                    HotelId = s.HotelId,
+                                    IsActive = s.IsActive,
+                                    Name = s.Name,
+                                    Price = s.Price,
+                                })
+                            };*/
+
+            // 1 query but have to fetch all ExtraServices info and select in-memory
+
+            var hotelViews =
+                from hotel in _dataContext.Hotels
+                join city in _dataContext.Cities
+                    on hotel.CityId equals city.CityId
+                join country in _dataContext.Countries
+                    on hotel.CountryId equals country.CountryId
+                where hotel.HotelId == query.HotelId
+                join service in _dataContext.ExtraServices
+                    on hotel.HotelId equals service.HotelId into services
+                from service in services.DefaultIfEmpty()
+                group service by new
+                {
+                    hotel.HotelId,
+                    hotel.Name,
+                    hotel.Address,
+                    CountryName = country.Name,
+                    CityName = city.Name,
+                    hotel.IsActive,
+                }
+                into gr
+                select gr;
+            var hotelGr = await hotelViews.FirstOrDefaultAsync();
+            return new HotelDetailsView
```

Comment:
isn't it possible to select HotelDetailsView as a part of query and then apply FirstOrDefaultAsync?